01_classify_stratified/nbconverted/plot_cancer_type_results.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Plot cancer type prediction results
+
+# In[1]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+from adjustText import adjust_text
+
+import mpmp.config as cfg
+import mpmp.utilities.analysis_utilities as au
+
+
+# In[2]:
+
+
+# set results directory
+results_dir = Path(cfg.results_dir, 'cancer_type').resolve()
+
+
+# In[3]:
+
+
+# load raw data
+results_df = au.load_cancer_type_prediction_results(results_dir, 'cancer_type')
+print(results_df.shape)
+results_df.head()
+
+
+# In[4]:
+
+
+# set this variable to filter plot to certain cancer types
+# if not included, plot all 33 of them
+# filter_cancer_types = ['LUAD', 'LUSC', 'THCA']
+filter_cancer_types = None
+if filter_cancer_types is not None:
+    filtered_df = results_df[
+        (results_df.signal == 'signal') &
+        (results_df.data_type == 'test') &
+        (results_df.cancer_type.isin(filter_cancer_types))
+    ]
+else:
+    filtered_df = results_df[
+        (results_df.signal == 'signal') &
+        (results_df.data_type == 'test')
+    ].sort_values(by='cancer_type')
+
+sns.set({'figure.figsize': (20, 5)})
+sns.boxplot(data=filtered_df, x='cancer_type', y='aupr', hue='training_data',
+            hue_order=np.sort(filtered_df.training_data.unique()))
```

Comment:
I'm pretty sure it's not sorted by default since I'm using the built-in pandas `Series.unique()` function: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.unique.html

You might be confusing this with the `numpy.unique()` function, which does sort by default: https://numpy.org/doc/stable/reference/generated/numpy.unique.html . I've definitely mixed these up in the past!

Reply:
Oh! I think I have to review a couple of lines in my code haha

Reply:
Yeah, it's definitely annoying/counterintuitive that they're not consistent, but such is life I guess.